src/core/api/link_raw_api.cpp

```diff
@@ -766,35 +769,83 @@ uint16_t otLinkGetShortAddress(otInstance *aInstance)
 }
 
 #if OPENTHREAD_ENABLE_DIAG
+static otInstance *sDiagInstance;
+
 void otDiagInit(otInstance *aInstance)
 {
-    (void)aInstance;
+    sDiagInstance = aInstance;
 }
 
 void otDiagProcessCmdLine(const char *aString, char *aOutput, size_t aOutputMaxLen)
 {
-    (void)aString;
-    (void)aOutput;
-    (void)aOutputMaxLen;
+    enum
+    {
+        kMaxArgs          = OPENTHREAD_CONFIG_DIAG_CMD_LINE_ARGS_MAX,
+        kMaxCommandBuffer = OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE,
+    };
+
+    otError error = OT_ERROR_NONE;
+    char    buffer[kMaxCommandBuffer];
+    char *  argVector[kMaxArgs];
+    uint8_t argCount = 0;
+
+    VerifyOrExit(strnlen(aString, kMaxCommandBuffer) < kMaxCommandBuffer, error = OT_ERROR_NO_BUFS);
+
+    strcpy(buffer, aString);
+    SuccessOrExit(error = Utils::CmdLineParser::ParseCmd(buffer, argCount, argVector, kMaxArgs));
+    VerifyOrExit(argCount >= 1, error = OT_ERROR_INVALID_ARGS);
+
+    if (strcmp(argVector[0], "power") == 0)
+    {
+        char * endptr;
+        int8_t power;
+
+        VerifyOrExit(argCount == 2, error = OT_ERROR_INVALID_ARGS);
+        power = static_cast<int8_t>(strtol(argVector[1], &endptr, 0));
+        VerifyOrExit(*endptr == '\0', error = OT_ERROR_INVALID_ARGS);
+
+        otPlatDiagTxPowerSet(power);
+    }
```

Comment:
How about doing `"channel"` as well (passing it through to `otPlatDiagChannelSet`)?